<!--
  FOR GAZEBO SIMULATION ONLY


  Launch file to execute two drivers for hong and kong in different namespaces
  using ros_control.

  Also launches the moveit planning execution and rviz to plan and execute
  trajectories.

  Adjust the ip addresses of the two robots in the argument section. Note that
  the choice of prefix owes to that used in combined urdf for the two robots.
  The namespace choice should be reflected in controller.yaml in
  ur10_dual_arm_moveit_config/config.

  Note to self: Check to see if joint state remapping required
-->

<launch>


    <!-- Including File ...upload.launch: sets robot_description parameter -->
    <include file="$(find ur10_cm)/launch/dual_arm_hong_kong_gazebo_upload.launch"/>



    <!-- Initialize and Spawn Gazebo -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
      <arg name="paused" value="false"/>
      <arg name="use_sim_time" value="true"/>
      <arg name="gui" value="false"/>
      <arg name="recording" value="false"/>
      <arg name="debug" value="false"/>
    </include>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model dual_arm_hong_kong" />

    <node name="publish_joint_states" pkg="ur10_cm" type="publish_joint_states.py"/>




    <!-- Arguments for robot hong -->
    <arg name="prefix_hong" default="hong_" />
    <!-- Arguments for robot kong-->
    <arg name="prefix_kong" default="kong_" />


    <group ns= "ns_hong">

      <remap from="joint_states" to="/joint_states"/>

      <rosparam file="$(find ur10_cm)/config/hong_dual_arm_controllers.yaml" command="load"/>

      <node name="ros_control_controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
      output="screen" args="hong_joint_state_controller hong_pos_based_pos_traj_controller --shutdown-timeout 50" />

      <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>



    </group>


    <group ns= "ns_kong">

      <remap from="joint_states" to="/joint_states"/>


      <rosparam file="$(find ur10_cm)/config/kong_dual_arm_controllers.yaml" command="load"/>

      <node name="ros_control_controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
      output="screen" args="kong_joint_state_controller kong_pos_based_pos_traj_controller --shutdown-timeout 50" />


      <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

    </group>


    <param name="trajectory_execution/allowed_start_tolerance" value="0.0"/>
    <include file="$(find ur10_dual_arm_gazebo_moveit_config)/launch/ur10_dual_arm_moveit_planning_execution.launch"/>
    <include file="$(find ur10_dual_arm_gazebo_moveit_config)/launch/moveit_rviz.launch"/>



</launch>
